marlowe-test/src/Spec/Marlowe/Semantics/Next/Contract/Generator.hs

```diff
@@ -0,0 +1,111 @@
+
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+
+module Spec.Marlowe.Semantics.Next.Contract.Generator
+  ( anyCaseContractsWithChoiceOnTheSameChoiceIdAndNonEmptyBounds
+  , anyCaseContractsWithChoiceOnlyNotShadowed
+  , anyCaseContractsWithEmptyBoundsChoiceOnly
+  , anyCaseContractsWithIdenticalEvaluatedDeposits
+  , anyCaseContractsWithoutIdenticalEvaluatedDeposits
+  , anyCloseOrReducedToAClose
+  , anyEmptyWhenNonTimedOut
+  , anyIrreducibleContract
+  , anyOnlyFalsifiedNotifies
+  , anyReducibleContract
+  , anyWithAtLeastOneNotifyTrue
+  , anyWithValidEnvironement
+  ) where
+
+
+import Language.Marlowe
+  (Action(Choice, Notify), Bound(Bound), Case(Case), ChoiceId(ChoiceId), Contract(Close, When), Environment, State)
+import Spec.Marlowe.Semantics.Arbitrary ()
+import Test.QuickCheck (Arbitrary(arbitrary), Gen, listOf, listOf1, suchThat)
+
+import Data.List (nub)
+import Spec.Marlowe.Semantics.Next.Common.Tuple (uncurry3)
+import Spec.Marlowe.Semantics.Next.Contract
+  (hasValidEnvironement, isEmptyWhenNonTimedOut, isIrreducible, isNotClose, isReducible, isReducibleToClose)
+import Spec.Marlowe.Semantics.Next.Contract.When.Deposit
+import Spec.Marlowe.Semantics.Next.Contract.When.Notify (areOnlyFalsifiedNotifies, atLeastOneNotifyTrue)
+
+
+anyReducibleContract :: Gen (Environment,State,Contract)
+anyReducibleContract
```

Comment:
These generators will all be very slow because of the pervasive use of `suchThat`